source/exe/main_common.cc

```diff
@@ -93,6 +93,18 @@ bool MainCommonBase::run() {
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+void MainCommonBase::adminRequest(absl::string_view path_and_query, absl::string_view method,
+                                  const AdminRequestFn& handler) {
```

Comment:
Rather than just exposing the admin interface for querying is there any objection to exposing the entire server interface through `MainCommonBase`?  For instance, this would allow containing code to directly query stats, add new admin handlers at runtime, and make timers/posts to Envoy's main thread among other things.  The server interface seems like a natural place to build an API boundary into Envoy.

Reply:
There has to be some place where we queue up requests to run on the main thread, and this seems like a good place to do that.

If we expose the Server& then it seems like it would be too easy to make a direct call to it from a different thread -- say the thread for handling an http request from a downstream console. But I'm open to suggestions.